src/sagemaker/jumpstart/session_utils.py

```diff
@@ -0,0 +1,203 @@
+        ),
+        inference_component_name,
+    )
+
+
+def _get_model_id_version_from_inference_component_endpoint_with_inference_component_name(
+    inference_component_name: str, sagemaker_session: Session
+):
+    """Returns the model ID and version inferred from a sagemaker inference component.
+
+    Raises:
+        ValueError: If the inference component does not have tags from which the model ID
+            and version can be inferred.
+    """
+    region: str = sagemaker_session.boto_region_name
+    partition: str = aws_partition(region)
+    account_id: str = sagemaker_session.account_id()
+
+    inference_component_arn = (
+        f"arn:{partition}:sagemaker:{region}:{account_id}:"
+        f"inference-component/{inference_component_name}"
+    )
+
+    model_id, model_version = get_jumpstart_model_id_version_from_resource_arn(
+        inference_component_arn, sagemaker_session
+    )
+
+    if not model_id:
+        raise ValueError(
+            "Cannot infer JumpStart model ID from inference component "
+            f"'{inference_component_name}'. Please specify JumpStart `model_id` "
+            "when retrieving default predictor for this inference component."
+        )
+
+    return model_id, model_version
+
+
+def _get_model_id_version_from_model_based_endpoint(
+    endpoint_name: str,
+    inference_component_name: Optional[str],
+    sagemaker_session: Session,
+) -> Tuple[str, str]:
+    """Returns the model ID and version inferred from a model-based endpoint.
+
+    Raises:
+        ValueError: If an inference component name is supplied, or if the endpoint does
+            not have tags from which the model ID and version can be inferred.
+    """
+
+    if inference_component_name:
+        raise ValueError("Cannot specify inference component name for model-based endpoints.")
+
+    region: str = sagemaker_session.boto_region_name
+    partition: str = aws_partition(region)
+    account_id: str = sagemaker_session.account_id()
+
+    # SageMaker Tagging requires endpoint names to be lower cased
+    endpoint_name = endpoint_name.lower()
+
+    endpoint_arn = f"arn:{partition}:sagemaker:{region}:{account_id}:endpoint/{endpoint_name}"
+
+    model_id, model_version = get_jumpstart_model_id_version_from_resource_arn(
+        endpoint_arn, sagemaker_session
+    )
+
+    if not model_id:
+        raise ValueError(
+            f"Cannot infer JumpStart model ID from endpoint '{endpoint_name}'. "
+            "Please specify JumpStart `model_id` when retrieving default "
+            "predictor for this endpoint."
+        )
+
+    return model_id, model_version
+
+
+def get_model_id_version_from_training_job(
+    training_job_name: str,
+    sagemaker_session: Optional[Session] = DEFAULT_JUMPSTART_SAGEMAKER_SESSION,
+) -> Tuple[str, str]:
+    """Returns the model ID and version inferred from a training job.
+
+    Raises:
+        ValueError: If the training job does not have tags from which the model ID
```

Comment:
Is there a doc link or more information to add about which tags are needed and how to add them?